backend/src/middleware/verifyJWT.ts

```diff
@@ -18,12 +22,14 @@ export const verifyJWT = (req: Request, res: Response, next: NextFunction) => {
   jwt.verify(
     token,
     process.env.REFRESH_TOKEN_SECRET as string,
-    (err: VerifyErrors | null, decoded: JwtPayload | string | undefined) => {
-      if (typeof decoded !== "string" && decoded?.userName) {
-        if (err) return res.sendStatus(403);
-        req.body.user = decoded.userName;
-        next();
-      }
+    (
+      err: jwt.VerifyErrors | null,
+      decoded: string | jwt.JwtPayload | undefined
+    ) => {
+      if (err) return res.sendStatus(500);
+      const decodedPayload = decoded as JwtPayloadWithUsername;
+      req.body.username = decodedPayload.username;
```

Comment:
sometimes we may want to let user change it's username like twice. then the only unique thing is user_id. please add this claim to the token, not the username.